spark/src/main/scala/io/substrait/spark/logical/ToLogicalPlan.scala

```diff
@@ -250,12 +250,21 @@ class ToLogicalPlan(spark: SparkSession) extends DefaultRelVisitor[LogicalPlan]
       case a: Aggregate => (a.aggregateExpressions, false)
       case other => (other.output, true)
     }
+    val names = if (project.getHint.isPresent) {
+      project.getHint.get().getOutputNames.asScala
+    } else {
+      List.empty
+    }
 
     withOutput(output) {
-      val projectList =
+      val projectExprs =
         project.getExpressions.asScala
           .map(expr => expr.accept(expressionConverter))
-          .map(toNamedExpression)
+      val projectList = if (names.size == projectExprs.size) {
```

Comment:
ah so this just skips using the names if there are inner structs? I guess that works, for a bit more complete solution I think you could use ToSparkType.toStructType to construct the expected schema (like [here](https://github.com/substrait-io/substrait-java/blob/d73723274f46ad6329fa97874788d3255ff3cd7a/spark/src/main/scala/io/substrait/spark/logical/ToLogicalPlan.scala#L424)) and then pick the column names from there, and for even more complete do that + reuse the `renameAndCastExprs`.